sdk/eventhub/azure-eventhubs/azure/eventhub/_client_base.py

```diff
@@ -66,15 +70,16 @@ def _parse_conn_str(conn_str, kwargs):
             "Invalid connection string. Should be in the format: "
             "Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>")
     entity = eventhub_name or entity_path
-    left_slash_pos = endpoint.find("//")
+    left_slash_pos = cast(str, endpoint).find("//")
     if left_slash_pos != -1:
-        host = endpoint[left_slash_pos + 2:]
+        host = cast(str, endpoint)[left_slash_pos + 2:]
     else:
-        host = endpoint
-    return host, shared_access_key_name, shared_access_key, entity
+        host = cast(str, endpoint)
+    return cast(str, host), cast(str, shared_access_key_name), cast(str, shared_access_key), cast(str, entity)
```

Comment:
I'm surprised you need that many cast. especially since line 77 you already cast host as str (for instance).